### ISSUE TITLE
Undoing reusable blocks to regular block conversion triggers a JS error

### ISSUE THREAD

**Issue by youknowriad** [WordPress/gutenberg#28519]
## Steps to reproduce.

 - Adds a reusable block to your post
 - Click "convert to regular blocks" button in the toolbar
 - Undo 
 - The editor breaks (JS error in useSelect call of `UncontrolledInnerBlocks`)

I tried guarding against undefined blocks in `UncontrolledInnerBlocks` but it seems this is not sufficient  because the  reusable block doesn't behave properly with  that (see the toolbar).

**Comment by talldan** [comment]
@youknowriad Is it the same error as reported in https://github.com/WordPress/gutenberg/issues/26421?

**Comment by youknowriad** [comment]
I'm not sure, It seems a bit different.

**Comment by strarsis** [comment]
+1, similar JS error:
````
TypeError: Cannot read property 'attributes' of null
    at [...]/app/plugins/block-options/build/index.js?ver=1614302145:16:192655
    at renderWithHooks ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:14938:20)
    at mountIndeterminateComponent ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:17617:15)
    at beginWork ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:18731:18)
    at HTMLUnknownElement.callCallback ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:182:16)
    at Object.invokeGuardedCallbackDev ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:231:18)
    at invokeGuardedCallback ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:286:33)
    at beginWork$1 ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:23338:9)
    at performUnitOfWork ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:22289:14)
    at workLoopSync ([...]/app/plugins/gutenberg/vendor/react-dom.82e849f1.js:22265:24)
````

**Comment by Mamaduka** [comment]
I'm no longer able to reproduce the issue on the current Gutenberg plugin and WP 5.9.

Based on @fluiddot's [investigation](https://github.com/WordPress/gutenberg/issues/32699#issuecomment-863906927) this probably got fixed after `getBlock` improvements - #34241.

https://user-images.githubusercontent.com/240569/163164418-66952929-92d4-4f6d-8f5b-25a827c2efe9.mp4
